### PR TITLE
Move factories to io.jenkins.blueocean.rest.factory in blueocean-rest

### PR DESCRIPTION
**Pull request by i386**
# Description

@vivek we can avoid the creation of `blueocean-rest-support` by moving factories to `io.jenkins.blueocean.rest.factory` package within `blueocean-rest` then ensure that we setup enforcer rules (as demonstrated in #971) to prevent any import leakage.

This means we can cut down on adding yet another additional module and split out  the `io.jenkins.blueocean.rest.factory` package later when we need to.

See [JENKINS-41281](https://issues.jenkins-ci.org/browse/JENKINS-41281).

Depends on https://github.com/jenkinsci/blueocean-plugin/pull/971

# Submitter checklist
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [x] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.
- [ ] Ran Acceptance Test Harness against PR changes.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

File: blueocean-rest/pom.xml [hunk 1]
```diff
@@ -24,4 +24,68 @@
...
+                                    <allowedImports>
+                                        <allowedImport>hudson.Extension</allowedImport>
+                                        <allowedImport>hudson.ExtensionPoint</allowedImport>
+                                        <allowedImport>hudson.ExtensionList</allowedImport>
+                                        <allowedImport>hudson.ExtensionListListener</allowedImport>
+                                        <allowedImport>hudson.security.csrf.CrumbExclusion</allowedImport>
+                                        <allowedImport>hudson.util.AdaptedIterator</allowedImport>
+                                    </allowedImports>
+                                    <excludedClasses>
+                                        <excludedClass>io.jenkins.blueocean.rest.factory.*</excludedClass>
```
Comment (i386) [inline blueocean-rest/pom.xml]: Excludes the `.factory` package from this strict enforcer rule

File: blueocean-rest/pom.xml [hunk 2]
```diff
@@ -24,4 +24,68 @@
...
+                                    </excludedClasses>
+                                </restrictImports>
+                                <!-- Factory package is allowed to import Jenkins models -->
+                                <restrictImports implementation="de.skuzzle.enforcer.restrictimports.RestrictImports">
```
Comment (i386) [inline blueocean-rest/pom.xml]: Allow the `.factory` package to import the packages in the enforcer rule above and only very particular model classes.

File: blueocean-rest/pom.xml (hunk 1 above)
Comment (vivek) [inline blueocean-rest/pom.xml]: Thats like wildcard import of jenkins models:)  Not sure with this change there is any point in keeping initial design criteria to keep rest model free of Jenkins core model for future proofing.

**Comment by i386** [comment]
ATH https://ci.blueocean.io/job/ATH-Jenkinsfile/job/master/2144/

**Comment by vivek** [comment]
LGTM 🐝

**Pull request merged**